From c08ac4ed4a8f149d7cae86eb4b862bbb57a4b905 Mon Sep 17 00:00:00 2001
From: Robert Bradshaw <robertwb@gmail.com>
Date: Mon, 26 Nov 2012 23:54:50 -0800
Subject: [PATCH 3/6] Fix compiler crash.

---
 Cython/Compiler/Nodes.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cython/Compiler/Nodes.py b/Cython/Compiler/Nodes.py
index c4f9f0b..6ba0e12 100644
--- a/Cython/Compiler/Nodes.py
+++ b/Cython/Compiler/Nodes.py
@@ -828,7 +828,7 @@ class CSimpleBaseTypeNode(CBaseTypeNode):
                 scope = env
                 for item in self.module_path:
                     entry = scope.lookup(item)
-                    if entry.is_cpp_class:
+                    if entry is not None and entry.is_cpp_class:
                         scope = entry.type.scope
                     else:
                         scope = None
-- 
1.8.0

