From 37724cccf7258445d4d5b710519fbf4bcc2df7dc Mon Sep 17 00:00:00 2001
From: Stefan Behnel <stefan_ml@behnel.de>
Date: Wed, 28 Nov 2012 21:15:28 +0100
Subject: [PATCH 5/6] fix 'cimport cython' in string code fragments

---
 Cython/Compiler/TreeFragment.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cython/Compiler/TreeFragment.py b/Cython/Compiler/TreeFragment.py
index 29ff12d..9832959 100644
--- a/Cython/Compiler/TreeFragment.py
+++ b/Cython/Compiler/TreeFragment.py
@@ -26,7 +26,7 @@ class StringParseContext(Main.Context):
         self.module_name = name
 
     def find_module(self, module_name, relative_to = None, pos = None, need_pxd = 1):
-        if module_name != self.module_name:
+        if module_name not in (self.module_name, 'cython'):
             raise AssertionError("Not yet supporting any cimports/includes from string code snippets")
         return ModuleScope(module_name, parent_module = None, context = self)
 
-- 
1.8.0

