Description: stop python3.4 from running a disabled test
Bug: https://mail.python.org/pipermail/cython-devel/2014-January/003939.html
Origin: https://github.com/cython/cython/commit/f7172efd729a6261c03502d0032ac1ac40dd3572

--- a/tests/memoryview/numpy_memoryview.pyx
+++ b/tests/memoryview/numpy_memoryview.pyx
@@ -406,7 +406,7 @@ cdef class DeallocateMe(object):
 # Disabled! References cycles don't seem to be supported by NumPy
 # @testcase
 def acquire_release_cycle(obj):
-    """
+    DISABLED_DOCSTRING = """
     >>> a = np.arange(20, dtype=np.object)
     >>> a[10] = DeallocateMe()
     >>> acquire_release_cycle(a)
